libsolidity/formal/SymbolicState.cpp

```diff
@@ -77,6 +77,10 @@ void SymbolicState::reset()
 	m_state.reset();
 	m_tx.reset();
 	m_crypto.reset();
+	/// We don't reset m_abi's pointer nor clear m_abiMembers on purpose,
+	/// since it only helps to keep the already generated types.
+	if (m_abi)
+		m_abi->reset();
```

Comment:
I'd maybe even just go for
```suggestion
	solAssert (m_abi, "");
	m_abi->reset();
```
Will it ever not be initialized here? And resetting and recreating ``m_abi`` here won't work anyways, will it? You *need* the correct sorts in it and it'd be hard to regenerate them after the reset, right?

So I'm basically saying (since you were wondering if this is ok and good): I think it's fine.